src/Illuminate/View/Mix/Mix.php

```diff
@@ -0,0 +1,271 @@
+     */
+    protected $manifestFilename = '/mix-manifest.json';
+
+    /**
+     * Get the path to a versioned Mix asset or a simple message if mix is disabled.
+     *
+     * @param  string  $path
+     * @param  string  $manifestDirectory
+     * @return \Illuminate\Support\HtmlString
+     */
+    public function resolve($path, $manifestDirectory = '')
+    {
+        if ($this->disabled) {
+            return $this->disabledPath();
+        }
+
+        return $this->getRealPath(
+            $this->sanitize($path),
+            $this->sanitize($manifestDirectory)
+        );
+    }
+
+    /**
+     * Get a message instead of the path when mix is disabled.
+     *
+     * @return \Illuminate\Support\HtmlString
+     */
+    protected function disabledPath()
+    {
+        return new HtmlString('Mix is disabled!');
+    }
+
+    /**
+     * Get the path to a versioned Mix file.
+     *
+     * @param  string  $path
+     * @param  string  $manifestDirectory
+     * @return \Illuminate\Support\HtmlString
+     */
+    protected function getRealPath($path, $manifestDirectory)
+    {
+        if ($this->hmrModeEnabled($manifestDirectory)) {
+            return $this->getHmrPath($path);
+        }
+
+        return $this->getCompiledPath($manifestDirectory, $path);
+    }
+
+    /**
+     * Get a sanitized version of a path.
+     *
+     * @param  string  $path
+     * @return string
+     */
+    protected function sanitize($path)
+    {
+        if (! Str::startsWith($path, '/')) {
+            $path = "/{$path}";
+        }
+
+        return $path;
+    }
+
+    /**
+     * Check if the HRM mode of Mix is enabled.
+     *
+     * @param  string  $manifestDirectory
+     *
+     * @return bool
+     */
+    protected function hmrModeEnabled($manifestDirectory)
+    {
+        return file_exists(public_path($manifestDirectory.$this->hmrFilename));
+    }
+
+    /**
+     * Get the full path to the file through the HMR server.
+     *
+     * @param  string  $path
+     *
+     * @return \Illuminate\Support\HtmlString
+     */
+    protected function getHmrPath($path)
+    {
+        return new HtmlString($this->hmrURI.$path);
+    }
+
+    /**
+     * Get the full path to the compiled file.
+     *
+     * @param  string  $manifestDirectory
+     * @param  string  $path
+     *
+     * @return \Illuminate\Support\HtmlString
+     */
+    protected function getCompiledPath($manifestDirectory, $path)
+    {
+        return new HtmlString($manifestDirectory.$this->getPathFromManifest($manifestDirectory, $path));
+    }
+
+    /**
+     * Get the path from the manifest file.
+     *
+     * @param  string  $manifestDirectory
+     * @param  string  $path
+     *
+     * @return string
+     * @throws \Illuminate\View\Mix\MixException
+     */
+    protected function getPathFromManifest($manifestDirectory, $path)
+    {
+        $manifest = $this->getManifest($manifestDirectory);
+
+        if (array_key_exists($path, $manifest)) {
```

Comment:
Can we change this to use ``` isset ``` instead? It would improve the performance a little bit...
It seems that ``` null ``` shouldn't be an expected value here anyway so the change in behavior should'nt matter...
Same for the one below.